tower-util/src/make_connection.rs

```diff
@@ -0,0 +1,42 @@
+use futures::Future;
+use tokio_io::{AsyncRead, AsyncWrite};
+use tower_service::Service;
+
+/// The MakeConnection trait is used to create transports
+///
+/// The goal of this service is to allow composable methods for creating
+/// `AsyncRead + AsyncWrite` transports. This could mean creating a TLS
+/// based connection or using some other method to authenticate the connection.
+pub trait MakeConnection<Request> {
+    /// The transport provided by this service
+    type Response: AsyncRead + AsyncWrite;
+
+    /// Errors produced by the connecting service
+    type Error;
+
+    /// The future that eventually produces the transport
+    type Future: Future<Item = Self::Response, Error = Self::Error>;
+
+    /// Connect and return a transport asynchronously
+    fn connect(&mut self, target: Request) -> Self::Future;
```

Comment:
Could this be renamed `make_connection`?

Reply:
done!